packages/atomic/src/components/common/result-templates/result-template-common.tsx

```diff
@@ -14,6 +17,36 @@ interface ResultTemplateCommonProps {
   setError: (error: Error) => void;
 }
 
+export function getTemplateNodeType(node: Node) {
+  if (isResultSectionNode(node)) {
+    return 'section';
+  }
+  if (!isVisualNode(node)) {
+    return 'metadata';
+  }
+  if (
+    isElementNode(node) &&
+    node.tagName.toLowerCase() === tableElementTagName
+  ) {
+    return 'table-column-definition';
+  }
+  return 'other';
+}
+
+function separateSectionAndNonSectionNodes(nodes: NodeList) {
```

Comment:
Suggestion:
```suggestion
function groupNodeByType(nodes: NodeList) {
```

Which would return an object where the keys are the `nodeType`.
```
{
'section': [...nodes],
'other': [...nodes],
'metadata': [...nodes],
'table-column-definition': [...nodes]
}
```

And then you could just deconstruct at the caller level.
Would that work ?

Reply:
Yes! Will do that.